dask/array/fft.py

```diff
@@ -86,19 +89,61 @@ def _ihfft_out_chunks(a, n, axis):
     return chunks
 
 
-fft = _fft_wrap(npfft.fft, np.complex_, _fft_out_chunks)
-
-
-ifft = _fft_wrap(npfft.ifft, np.complex_, _fft_out_chunks)
-
+def fft_wrapper(fft_func, kind):
+    """ Wrapper of 1D complex FFT functions
 
-rfft = _fft_wrap(npfft.rfft, np.complex_, _rfft_out_chunks)
+    Takes a function that behaves like ``numpy.fft`` functions and
+    a specified kind to match it to that are named after the functions
+    in the ``numpy.fft`` API.
 
+    Supported kinds include:
 
-irfft = _fft_wrap(npfft.irfft, np.float_, _irfft_out_chunks)
-
-
-hfft = _fft_wrap(npfft.hfft, np.float_, _hfft_out_chunks)
+        * fft
+        * ifft
+        * rfft
+        * irfft
+        * hfft
+        * ihfft
 
+    >>> fft = fft_wrapper(npfft.fft, kind="fft")
+    """
 
-ihfft = _fft_wrap(npfft.ihfft, np.complex_, _ihfft_out_chunks)
+    params = {
+        "fft": {
+            "dtype": np.complex_,
+            "out_chunk_fn" : _fft_out_chunks
+        },
+        "ifft": {
+            "dtype": np.complex_,
+            "out_chunk_fn" : _fft_out_chunks
+        },
+        "rfft": {
+            "dtype": np.complex_,
+            "out_chunk_fn" : _rfft_out_chunks
+        },
+        "irfft": {
+            "dtype": np.float_,
+            "out_chunk_fn" : _irfft_out_chunks
+        },
+        "hfft": {
+            "dtype": np.float_,
+            "out_chunk_fn" : _hfft_out_chunks
+        },
+        "ihfft": {
+            "dtype": np.complex_,
+            "out_chunk_fn" : _ihfft_out_chunks
+        },
+    }
```

Comment:
The intent behind the data comment is to keep this data outside of functions.  That way it can be used by other functions as well.

```python
dtypes = {'fft: ...}
funcs = {'fft': ...

def func(name, ...):
    dtype = dtypes[name]
    chunk_func = funcs[name]
```